app/screens/redeem-lnurl-withdrawal-screen/redeem-bitcoin-result-screen.tsx

```diff
@@ -0,0 +1,388 @@
+    lnUpdate?.status === "PAID"
+
+  const [lnInvoiceCreate] = useLnInvoiceCreateMutation()
+  const [lnUsdInvoiceCreate] = useLnUsdInvoiceCreateMutation()
+
+  const createWithdrawRequestInvoice = useCallback(
+    async ({ satAmount, memo }) => {
+      setInvoice(null)
+      try {
+        if (type === TYPE_LIGHTNING_BTC) {
+          logGeneratePaymentRequest({
+            paymentType: "lightning",
+            hasAmount: true,
+            receivingWallet: WalletCurrency.Btc,
+          })
+          const {
+            data: {
+              lnInvoiceCreate: { invoice, errors },
+            },
+          } = await lnInvoiceCreate({
+            variables: {
+              input: { walletId, amount: satAmount, memo },
+            },
+          })
+          if (errors && errors.length !== 0) {
+            console.error(errors, "error with lnInvoiceCreate")
+            setErr(LL.RedeemBitcoinScreen.error())
+            return
+          }
+
+          setInvoice(invoice)
+        } else {
+          logGeneratePaymentRequest({
+            paymentType: "lightning",
+            hasAmount: true,
+            receivingWallet: WalletCurrency.Usd,
+          })
+          const {
+            data: {
+              lnUsdInvoiceCreate: { invoice, errors },
+            },
+          } = await lnUsdInvoiceCreate({
+            variables: {
+              input: { walletId, amount: satAmountInUsd * 100, memo },
+            },
+          })
+
+          if (errors && errors.length !== 0) {
+            console.error(errors, "error with lnInvoiceCreate")
+            setErr(LL.ReceiveBitcoinScreen.error())
+            return
+          }
+          setInvoice(invoice)
+        }
+      } catch (err) {
+        console.error(err, "error with AddInvoice")
+        setErr(`${err}`)
+        throw err
+      }
+    },
+    [lnInvoiceCreate, lnUsdInvoiceCreate, LL],
+  )
+
+  const submitLNURLWithdrawRequest = async (generatedInvoice) => {
+    const url = `${callback}${callback.includes("?") ? "&" : "?"}k1=${k1}&pr=${
+      generatedInvoice.paymentRequest
+    }`
+
+    const result = await fetch(url)
+
+    if (result.ok) {
+      const lnurlResponse = await result.json()
+      if (lnurlResponse?.status?.toLowerCase() === "ok") {
+        // TODO: Set processing payment
+      } else {
+        console.error(lnurlResponse, "error with redeeming")
+        // TODO: Set failed payment
+        setErr(LL.RedeemBitcoinScreen.redeemingError())
+      }
+    } else {
+      console.error(result.text(), "error with submitting withdrawalRequest")
+      setErr(LL.RedeemBitcoinScreen.submissionError())
+    }
+  }
+
+  useEffect((): void | (() => void) => {
+    if (withdrawalInvoice) {
+      submitLNURLWithdrawRequest(withdrawalInvoice)
+    } else {
+      createWithdrawRequestInvoice({ satAmount, memo })
+    }
+  }, [walletId, withdrawalInvoice, memo, type, satAmount, createWithdrawRequestInvoice])
```

Comment:
In order to prevent missing dependencies in hooks, can you add submitLNURLWithdrawRequest (you might have to memoize it or use callback so that it doesnt keep rerendering). `type` and `walletId` aren't used in this hook, so I dont believe they should be dependencies here, but should be instead added as dependencies to `createWithdrawRequestInvoice`.